client/ayon_houdini/plugins/publish/validate_render_product_paths_unique.py

```diff
@@ -0,0 +1,147 @@
+    # output files than just a single file or single sequence of files.
+    expected_files: List[Dict[str, List[str]]] = (
+        instance.data.get("expectedFiles", [])
+    )
+    filepaths: List[str] = []
+    if expected_files:
+        # Products with expected files
+        # This can be Render products or submitted cache to farm.
+        for expected in expected_files:
+            # expected.values() is a list of lists
+            filepaths.extend(expected.values())
+    else:
+        # Products with frames or single file.
+        staging_dir = instance.data.get("stagingDir")
+        frames = instance.data.get("frames")
+        if frames is None or not staging_dir:
+            return []
+
+        if isinstance(frames, str):
+            # single file.
+            filepaths.append(f"{staging_dir}/{frames}")
+        else:
+            # list of frame.
+            filepaths.extend(
+                [f"{staging_dir}/{frame}" for frame in frames]
+            )
+
+    return filepaths
+
+
+class ValidateRenderProductPathsUnique(plugin.HoudiniContextPlugin,
+                                       OptionalPyblishPluginMixin):
+    """Validate that render product paths are unique.
+
+    This allows to catch before rendering whether multiple render ROPs would
+    end up writing to the same filepaths. This can be a problem when rendering
+    because each render job would overwrite the files of the other at
+    rendertime.
+
+    """
+    order = pyblish.api.ValidatorOrder
+    families = [
+        # Render products
+        "usdrender", "karma_rop", "redshift_rop", "arnold_rop", "mantra_rop",
+
+        # Product families from collect frames plug-in
+        "camera", "vdbcache", "imagesequence", "ass", "redshiftproxy",
+        "review", "pointcache", "fbx", "model"
+    ]
+
+    hosts = ["houdini"]
+    label = "Unique Render Product Paths"
+    actions = [SelectInvalidAction]
+    optional = True
+
+    def process(self, context):
+        if not self.is_active(context.data):
+            return
+
+        invalid = self.get_invalid(context)
+        if not invalid:
+            return
+
+        node_paths = [node.path() for node in invalid]
+        node_paths.sort()
+        invalid_list = "\n".join(f"- {path}" for path in node_paths)
+        raise PublishValidationError(
+            "Multiple instances render to the same path. "
+            "Please make sure each ROP renders to a unique output path:\n"
+            f"{invalid_list}",
+            title=self.label,
+            description=self.get_description()
+        )
+
+    @classmethod
+    def get_invalid(cls, context) -> "List[hou.Node]":
+        # Get instances matching this plugin families
+        instances = pyblish.api.instances_by_plugin(list(context), cls)
+        if len(instances) < 2:
+            return []
+
+        # Get expected rendered filepaths
+        paths_to_instance_id = defaultdict(list)
+        for instance in instances:
```

Comment:
This will make our lives easier although it might not be a common error 😅 

```suggestion
        for instance in instances:
            # Skip the original instance when local rendering and having runtime instances
            if instance.data.get("integrate") is False:
                continue 
```

Here's an example: 
I've a render rop with 3 separate AOVs. I duplicated the rop.
the first one is set to local rendering and the second set to farm. 
the log after applying my suggestion will be 
```
WARNING: Multiple instances output to path: //storage/work/ayon/RnD/features/deadline_submissions/work/Generic/pyblish/renders/mantra_ropTest/mantra_ropTest_beauty.%d.exr [1001-1005]
WARNING: Multiple instances output to path: //storage/work/ayon/RnD/features/deadline_submissions/work/Generic/pyblish/renders/mantra_ropTest/mantra_ropTest_N.%d.exr [1001-1005]
WARNING: Multiple instances output to path: //storage/work/ayon/RnD/features/deadline_submissions/work/Generic/pyblish/renders/mantra_ropTest/mantra_ropTest_P.%d.exr [1001-1005]
Traceback (most recent call last):
  File "C:\Users\Mustafa Taher\AppData\Local\Ynput\AYON\dependency_packages\ayon_2411052009_windows.zip\dependencies\pyblish\plugin.py", line 528, in __explicit_process
    runner(*args)
  File "E:\Ynput\ayon-houdini\client\ayon_houdini\plugins\publish\validate_render_product_paths_unique.py", line 88, in process
    raise PublishValidationError(
ayon_core.pipeline.publish.publish_plugins.PublishValidationError: Multiple instances render to the same path. Please make sure each ROP renders to a unique output path:
- /out/mantra_ropTest
- /out/mantra_ropTest1
- /out/mantra_ropTest1
- /out/mantra_ropTest1
```

Currently, (without my suggestion) it will be: (note that `/out/mantra_ropTest1` is mentioned 4 times (original + 3 aovs))
```
WARNING: Multiple instances output to path: //storage/work/ayon/RnD/features/deadline_submissions/work/Generic/pyblish/renders/mantra_ropTest/mantra_ropTest_beauty.%d.exr [1001-1005]
WARNING: Multiple instances output to path: //storage/work/ayon/RnD/features/deadline_submissions/work/Generic/pyblish/renders/mantra_ropTest/mantra_ropTest_N.%d.exr [1001-1005]
WARNING: Multiple instances output to path: //storage/work/ayon/RnD/features/deadline_submissions/work/Generic/pyblish/renders/mantra_ropTest/mantra_ropTest_P.%d.exr [1001-1005]
Traceback (most recent call last):
  File "C:\Users\Mustafa Taher\AppData\Local\Ynput\AYON\dependency_packages\ayon_2411052009_windows.zip\dependencies\pyblish\plugin.py", line 528, in __explicit_process
    runner(*args)
  File "E:\Ynput\ayon-houdini\client\ayon_houdini\plugins\publish\validate_render_product_paths_unique.py", line 88, in process
    raise PublishValidationError(
ayon_core.pipeline.publish.publish_plugins.PublishValidationError: Multiple instances render to the same path. Please make sure each ROP renders to a unique output path:
- /out/mantra_ropTest
- /out/mantra_ropTest1
- /out/mantra_ropTest1
- /out/mantra_ropTest1
- /out/mantra_ropTest1
```